funcx_sdk/funcx/sdk/asynchronous/ws_polling_task.py

```diff
@@ -119,24 +119,38 @@ async def send_outgoing(self, queue: asyncio.Queue):
             task_group_id = await queue.get()
             await self.ws.send(task_group_id)
 
-    async def handle_incoming(self, pending_futures, auto_close=False):
+    async def handle_incoming(self, pending_futures, auto_close=False) -> bool:
+        """
+
+        Parameters
+        ----------
+        pending_futures
+        auto_close
+
+        Returns
+        -------
+        True -- If connection is closing from internal shutdown process
+        False -- External disconnect - to be handled by reconnect logic
+        """
+
         while True:
             try:
                 raw_data = await asyncio.wait_for(self.ws.recv(), timeout=1.0)
             except asyncio.TimeoutError:
                 pass
             except ConnectionClosedOK:
                 if self.closed_by_main_thread:
-                    log.debug("WebSocket connection closed by main thread")
+                    log.info("WebSocket connection closed by main thread")
+                    return True
                 else:
-                    log.error("WebSocket connection closed unexpectedly")
-                return
+                    log.info("WebSocket connection closed unexpectedly")
+                    return False
```

Comment:
This is the only case, as far as I can tell, in which this returns `False`. Presumably this could instead just read `self.reconnect()` (which would need to be defined)?

Reply:
That's one way to do it. I think, in this case, it's cleaner to have the listener exit and have the layer above do the reconnect.